Compile error with Angular 10.1.2: `error TS2380: 'get' and 'set' accessor must have the same type`

# Bug Report

## Prerequisites

Please answer the following questions for yourself before submitting an issue.
**YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the Readme and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] This is related to angular-google-charts and not to Google Charts directly (if it's a Google Charts issue, [their forum](https://groups.google.com/forum/#!forum/google-visualization-api) may help)

## Description

After installing as per the instructions, and adding the `<google-chart>` component to the app, Angular `ng serve` spews:

```
    ERROR in node_modules/angular-google-charts/lib/components/chart-wrapper/chart-wrapper.component.d.ts:28:9 - error TS2380: 'get' and 'set' accessor must have the same type.
    
    28     get chartWrapper(): google.visualization.ChartWrapper | null;
               ~~~~~~~~~~~~
    node_modules/angular-google-charts/lib/components/chart-wrapper/chart-wrapper.component.d.ts:29:9 - error TS2380: 'get' and 'set' accessor must have the same type.
    
    29     set chartWrapper(wrapper: google.visualization.ChartWrapper);
               ~~~~~~~~~~~~
```

I'm using Angular 10.1.2 with Ivy.

Looking at the code, it indeed looks that `set chartWrapper` is missing `|null`.

## Your environment
<pre><code>
Angular CLI: 10.1.2
Node: 10.19.0
OS: linux x64

Angular: 10.1.2
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1001.2
@angular-devkit/build-angular   0.1001.2
@angular-devkit/core            10.1.2
@angular-devkit/schematics      10.1.2
@angular/cdk                    10.2.1
@angular/material               10.2.1
@schematics/angular             10.1.2
@schematics/update              0.1001.2
rxjs                            6.6.3
typescript                      4.0.2
</code></pre>

--- a/apps/playground/src/app/main/main.component.ts
+++ b/apps/playground/src/app/main/main.component.ts
@@ -44,7 +44,7 @@ export class MainComponent implements OnInit {
   };
 
   @ViewChild('chart', { static: true })
-  public chart: GoogleChartComponent;
+  public chart!: GoogleChartComponent;
 
   constructor(private router: Router) {
     this.charts.push({
@@ -189,7 +189,7 @@ export class MainComponent implements OnInit {
     this.charts.push({
       title: 'Candlestick Chart',
       type: ChartType.CandlestickChart,
-      columns: null,
+      columns: undefined,
       data: [
         ['Mon', 20, 28, 38, 45],
         ['Tue', 31, 38, 55, 66],

--- a/apps/playground/src/app/test/test.component.ts
+++ b/apps/playground/src/app/test/test.component.ts
@@ -54,7 +54,7 @@ export class TestComponent {
   };
 
   @ViewChild(ChartEditorComponent)
-  public readonly editor: ChartEditorComponent;
+  public readonly editor!: ChartEditorComponent;
 
   constructor(private location: Location) {}
 

--- a/libs/angular-google-charts/src/lib/components/chart-base/chart-base.component.ts
+++ b/libs/angular-google-charts/src/lib/components/chart-base/chart-base.component.ts
@@ -29,14 +29,16 @@ export interface ChartBase {
   select: EventEmitter<ChartSelectionChangedEvent>;
 
   /**
-   * The drawn chart or `null`.
+   * The drawn chart or `undefined`.
    */
-  chart: google.visualization.ChartBase | null;
+  chart: google.visualization.ChartBase | undefined;
 
   /**
-   * The underlying chart wrapper or `null`.
+   * The underlying chart wrapper.
+   *
+   * This will throw an exception when trying to access the chart wrapper before `wrapperReady$` emits.
    */
-  chartWrapper: google.visualization.ChartWrapper | null;
+  chartWrapper: google.visualization.ChartWrapper;
 
   /**
    * Emits after the `ChartWrapper` is created, but before the chart is drawn for the first time.

--- a/libs/angular-google-charts/src/lib/components/chart-editor/chart-editor.component.ts
+++ b/libs/angular-google-charts/src/lib/components/chart-editor/chart-editor.component.ts
@@ -15,7 +15,7 @@ import { ChartEditorRef } from './chart-editor-ref';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class ChartEditorComponent implements OnInit {
-  private editor: google.visualization.ChartEditor;
+  private editor: google.visualization.ChartEditor | undefined;
   private initializedSubject = new Subject<google.visualization.ChartEditor>();
 
   constructor(private scriptLoaderService: ScriptLoaderService) {}
@@ -45,6 +45,13 @@ export class ChartEditorComponent implements OnInit {
   public editChart(component: ChartBase): ChartEditorRef;
   public editChart(component: ChartBase, options: google.visualization.ChartEditorOptions): ChartEditorRef;
   public editChart(component: ChartBase, options?: google.visualization.ChartEditorOptions) {
+    if (!component.chartWrapper) {
+      throw new Error('Chart wrapper is `undefined`. Please wait for the `initialized$` observable before trying to edit a chart.');
+    }
+    if (!this.editor) {
+      throw new Error('Chart editor is `undefined`. Please wait for the `initialized$` observable before trying to edit a chart.');
+    }
+
     const handle = new ChartEditorRef(this.editor);
     this.editor.openDialog(component.chartWrapper, options || {});
 

--- a/libs/angular-google-charts/src/lib/components/chart-wrapper/chart-wrapper.component.spec.ts
+++ b/libs/angular-google-charts/src/lib/components/chart-wrapper/chart-wrapper.component.spec.ts
@@ -3,6 +3,7 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { EMPTY, of } from 'rxjs';
 
 import { ChartType } from '../../models/chart-type.model';
+import { ChartErrorEvent, ChartReadyEvent, ChartSelectionChangedEvent } from '../../models/events.model';
 import { ScriptLoaderService } from '../../script-loader/script-loader.service';
 
 import { ChartWrapperComponent } from './chart-wrapper.component';
@@ -82,7 +83,7 @@ describe('ChartWrapperComponent', () => {
     it('should not throw if the specs are `null`', () => {
       const scriptLoaderService = TestBed.inject(ScriptLoaderService) as jest.Mocked<ScriptLoaderService>;
       scriptLoaderService.loadChartPackages.mockReturnValue(of(null));
-      component.specs = null;
+      component.specs = undefined;
 
       expect(() => component.ngOnInit()).not.toThrow();
     });
@@ -146,7 +147,7 @@ describe('ChartWrapperComponent', () => {
     });
 
     it('should not throw if the wrapper is not yet initialized', () => {
-      component['wrapper'] = null;
+      component['wrapper'] = undefined;
       component['initialized'] = false;
 
       const specs = {
@@ -185,8 +186,8 @@ describe('ChartWrapperComponent', () => {
       expect(chartWrapperMock.setContainerId).not.toHaveBeenCalled();
     });
 
-    it('should not throw if the specs are `null`', () => {
-      expect(() => changeSpecs(null)).not.toThrow();
+    it('should not throw if the specs are `undefined`', () => {
+      expect(() => changeSpecs(undefined)).not.toThrow();
 
       expect(chartWrapperMock.draw).toHaveBeenCalled();
     });
@@ -213,13 +214,14 @@ describe('ChartWrapperComponent', () => {
 
   describe('chart', () => {
     it('should not throw when trying to access chart if its not yet drawn', () => {
+      component['wrapper'] = chartWrapperMock as any;
       expect(() => component.chart).not.toThrow();
     });
   });
 
   describe('chartWrapper', () => {
-    it('should not throw if the chart wrapper is `null`', () => {
-      expect(() => component.chartWrapper).not.toThrow();
+    it('should throw if the chart wrapper is `undefined`', () => {
+      expect(() => component.chartWrapper).toThrow();
     });
 
     it('should return the chart wrapper', () => {
@@ -249,7 +251,7 @@ describe('ChartWrapperComponent', () => {
       component.ngOnInit();
 
       const readySpy = jest.fn();
-      component.ready.subscribe(event => readySpy(event));
+      component.ready.subscribe((event: ChartReadyEvent) => readySpy(event));
 
       const readyCallback = visualizationMock.events.addListener.mock.calls[0][2];
       readyCallback();
@@ -261,7 +263,7 @@ describe('ChartWrapperComponent', () => {
       component.ngOnInit();
 
       const errorSpy = jest.fn();
-      component.error.subscribe(event => errorSpy(event));
+      component.error.subscribe((event: ChartErrorEvent) => errorSpy(event));
 
       const errorCallback = visualizationMock.events.addListener.mock.calls[1][2];
 
@@ -278,7 +280,7 @@ describe('ChartWrapperComponent', () => {
       chartWrapperMock.getChart.mockReturnValue(chartMock);
 
       const selectSpy = jest.fn();
-      component.select.subscribe(event => selectSpy(event));
+      component.select.subscribe((event: ChartSelectionChangedEvent) => selectSpy(event));
 
       component.ngOnInit();
 
@@ -292,7 +294,7 @@ describe('ChartWrapperComponent', () => {
     });
   });
 
-  function changeSpecs(newValue: google.visualization.ChartSpecs) {
+  function changeSpecs(newValue?: google.visualization.ChartSpecs) {
     const oldValue = component.specs;
     component.specs = newValue;
     component.ngOnChanges({ specs: new SimpleChange(oldValue, newValue, oldValue == null) });

--- a/libs/angular-google-charts/src/lib/components/chart-wrapper/chart-wrapper.component.ts
+++ b/libs/angular-google-charts/src/lib/components/chart-wrapper/chart-wrapper.component.ts
@@ -33,7 +33,7 @@ export class ChartWrapperComponent implements ChartBase, OnChanges, OnInit {
    * rendering the chart into the components' template.
    */
   @Input()
-  public specs: google.visualization.ChartSpecs;
+  public specs?: google.visualization.ChartSpecs;
 
   @Output()
   public error = new EventEmitter<ChartErrorEvent>();
@@ -44,25 +44,25 @@ export class ChartWrapperComponent implements ChartBase, OnChanges, OnInit {
   @Output()
   public select = new EventEmitter<ChartSelectionChangedEvent>();
 
-  private wrapper: google.visualization.ChartWrapper;
+  private wrapper: google.visualization.ChartWrapper | undefined;
   private wrapperReadySubject = new ReplaySubject<google.visualization.ChartWrapper>(1);
   private initialized = false;
 
   constructor(private element: ElementRef, private scriptLoaderService: ScriptLoaderService) {}
 
-  public get chart(): google.visualization.ChartBase | null {
-    if (!this.wrapper) {
-      return null;
-    }
-
-    return this.wrapper.getChart();
+  public get chart(): google.visualization.ChartBase | undefined {
+    return this.chartWrapper.getChart();
   }
 
   public get wrapperReady$() {
     return this.wrapperReadySubject.asObservable();
   }
 
-  public get chartWrapper(): google.visualization.ChartWrapper | null {
+  public get chartWrapper(): google.visualization.ChartWrapper {
+    if (!this.wrapper) {
+      throw new Error('Cannot access the chart wrapper before initialization.');
+    }
+
     return this.wrapper;
   }
 
@@ -112,18 +112,23 @@ export class ChartWrapperComponent implements ChartBase, OnChanges, OnInit {
       this.specs = {} as google.visualization.ChartSpecs;
     }
 
-    this.wrapper.setChartType(this.specs.chartType);
-    this.wrapper.setDataTable(this.specs.dataTable as any); // The typing here are not correct, this also accepts plain arrays
-    this.wrapper.setDataSourceUrl(this.specs.dataSourceUrl);
-    this.wrapper.setDataSourceUrl(this.specs.dataSourceUrl);
-    this.wrapper.setQuery(this.specs.query);
-    this.wrapper.setOptions(this.specs.options);
-    this.wrapper.setRefreshInterval(this.specs.refreshInterval);
-    this.wrapper.setView(this.specs.view);
+    // The typing here are not correct. These methods accept `undefined` as well.
+    // That's why we have to cast to `any`
+
+    this.wrapper!.setChartType(this.specs.chartType);
+    this.wrapper!.setDataTable(this.specs.dataTable as any);
+    this.wrapper!.setDataSourceUrl(this.specs.dataSourceUrl as any);
+    this.wrapper!.setDataSourceUrl(this.specs.dataSourceUrl as any);
+    this.wrapper!.setQuery(this.specs.query as any);
+    this.wrapper!.setOptions(this.specs.options as any);
+    this.wrapper!.setRefreshInterval(this.specs.refreshInterval as any);
+    this.wrapper!.setView(this.specs.view);
   }
 
   private drawChart() {
-    this.wrapper.draw();
+    if (this.wrapper) {
+      this.wrapper.draw();
+    }
   }
 
   private registerChartEvents() {
@@ -133,10 +138,10 @@ export class ChartWrapperComponent implements ChartBase, OnChanges, OnInit {
       google.visualization.events.addListener(object, eventName, callback);
     };
 
-    registerChartEvent(this.wrapper, 'ready', () => this.ready.emit({ chart: this.chart }));
+    registerChartEvent(this.wrapper, 'ready', () => this.ready.emit({ chart: this.chart! }));
     registerChartEvent(this.wrapper, 'error', (error: ChartErrorEvent) => this.error.emit(error));
     registerChartEvent(this.wrapper, 'select', () => {
-      const selection = this.chart.getSelection();
+      const selection = this.chart!.getSelection();
       this.select.emit({ selection });
     });
   }

--- a/libs/angular-google-charts/src/lib/components/control-wrapper/control-wrapper.component.spec.ts
+++ b/libs/angular-google-charts/src/lib/components/control-wrapper/control-wrapper.component.spec.ts
@@ -3,6 +3,7 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { EMPTY, of } from 'rxjs';
 
 import { FilterType } from '../../models/control-type.model';
+import { ChartErrorEvent, ChartReadyEvent } from '../../models/events.model';
 import { ScriptLoaderService } from '../../script-loader/script-loader.service';
 
 import { ControlWrapperComponent } from './control-wrapper.component';
@@ -113,7 +114,7 @@ describe('ControlWrapperComponent', () => {
   describe('ngOnChanges', () => {
     function changeInput<K extends keyof ControlWrapperComponent>(property: K, newValue: ControlWrapperComponent[K]) {
       const oldValue = component[property];
-      component[property as any] = newValue;
+      component[property] = newValue;
       component.ngOnChanges({ [property]: new SimpleChange(oldValue, newValue, oldValue == null) });
     }
 
@@ -124,7 +125,6 @@ describe('ControlWrapperComponent', () => {
     it('should update the control type if it changed', () => {
       const controlWrapperMock = { setControlType: jest.fn() };
       component['_controlWrapper'] = controlWrapperMock as any;
-      component['initialized'] = true;
 
       const type = FilterType.Category;
       changeInput('type', type);
@@ -135,7 +135,6 @@ describe('ControlWrapperComponent', () => {
     it('should update the options if they changed', () => {
       const controlWrapperMock = { setOptions: jest.fn() };
       component['_controlWrapper'] = controlWrapperMock as any;
-      component['initialized'] = true;
 
       const options = { key: 'value' };
       changeInput('options', options);
@@ -146,7 +145,6 @@ describe('ControlWrapperComponent', () => {
     it('should update the state if it changed', () => {
       const controlWrapperMock = { setState: jest.fn() };
       component['_controlWrapper'] = controlWrapperMock as any;
-      component['initialized'] = true;
 
       const state = { from: 'to' };
       changeInput('state', state);
@@ -168,7 +166,7 @@ describe('ControlWrapperComponent', () => {
 
     it('should emit ready event', () => {
       const readySpy = jest.fn();
-      component.ready.subscribe(event => readySpy(event));
+      component.ready.subscribe((event: ChartReadyEvent) => readySpy(event));
 
       // This leads to the component subscribing to all events
       component.ngOnInit();
@@ -183,7 +181,7 @@ describe('ControlWrapperComponent', () => {
 
     it('should emit error event', () => {
       const errorSpy = jest.fn();
-      component.error.subscribe(event => errorSpy(event));
+      component.error.subscribe((event: ChartErrorEvent) => errorSpy(event));
 
       // This leads to the component subscribing to all events
       component.ngOnInit();
@@ -198,7 +196,7 @@ describe('ControlWrapperComponent', () => {
 
     it('should emit statechange event', () => {
       const stateChangeSpy = jest.fn();
-      component.stateChange.subscribe(event => stateChangeSpy(event));
+      component.stateChange.subscribe((event: unknown) => stateChangeSpy(event));
 
       // This leads to the component subscribing to all events
       component.ngOnInit();

--- a/libs/angular-google-charts/src/lib/components/control-wrapper/control-wrapper.component.ts
+++ b/libs/angular-google-charts/src/lib/components/control-wrapper/control-wrapper.component.ts
@@ -13,7 +13,7 @@ import { ReplaySubject } from 'rxjs';
 
 import { generateRandomId } from '../../helpers/id.helper';
 import { FilterType } from '../../models/control-type.model';
-import { ChartErrorEvent } from '../../models/events.model';
+import { ChartErrorEvent, ChartReadyEvent } from '../../models/events.model';
 import { ScriptLoaderService } from '../../script-loader/script-loader.service';
 import { ChartBase } from '../chart-base/chart-base.component';
 
@@ -87,7 +87,7 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
    * and call control methods only after the event was fired.
    */
   @Output()
-  public ready = new EventEmitter<void>();
+  public ready = new EventEmitter<ChartReadyEvent>();
 
   /**
    * Emits when the user interacts with the control, affecting its state.
@@ -96,7 +96,7 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
    * To retrieve an updated control state after the event fired, call `ControlWrapper.getState()`.
    */
   @Output()
-  public stateChange = new EventEmitter<void>();
+  public stateChange = new EventEmitter<unknown>();
 
   /**
    * A generated id assigned to this components DOM element.
@@ -104,9 +104,8 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
   @HostBinding('id')
   public readonly id = generateRandomId();
 
-  private _controlWrapper: google.visualization.ControlWrapper;
+  private _controlWrapper?: google.visualization.ControlWrapper;
   private wrapperReadySubject = new ReplaySubject<google.visualization.ControlWrapper>(1);
-  private initialized = false;
 
   constructor(private loaderService: ScriptLoaderService) {}
 
@@ -117,19 +116,22 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
     return this.wrapperReadySubject.asObservable();
   }
 
-  public get controlWrapper(): google.visualization.ControlWrapper | null {
+  public get controlWrapper(): google.visualization.ControlWrapper {
+    if (!this._controlWrapper) {
+      throw new Error(`Cannot access the control wrapper before it being initialized.`);
+    }
+
     return this._controlWrapper;
   }
 
   public ngOnInit() {
     this.loaderService.loadChartPackages('controls').subscribe(() => {
       this.createControlWrapper();
-      this.initialized = true;
     });
   }
 
   public ngOnChanges(changes: SimpleChanges): void {
-    if (!this.initialized) {
+    if (!this._controlWrapper) {
       return;
     }
 
@@ -138,11 +140,11 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
     }
 
     if (changes.options) {
-      this._controlWrapper.setOptions(this.options);
+      this._controlWrapper.setOptions(this.options || {});
     }
 
     if (changes.state) {
-      this._controlWrapper.setState(this.state);
+      this._controlWrapper.setState(this.state || {});
     }
   }
 
@@ -161,8 +163,8 @@ export class ControlWrapperComponent implements OnInit, OnChanges {
   private addEventListeners() {
     google.visualization.events.removeAllListeners(this._controlWrapper);
 
-    google.visualization.events.addListener(this._controlWrapper, 'ready', event => this.ready.emit(event));
-    google.visualization.events.addListener(this._controlWrapper, 'error', event => this.error.emit(event));
-    google.visualization.events.addListener(this._controlWrapper, 'statechange', event => this.stateChange.emit(event));
+    google.visualization.events.addListener(this._controlWrapper, 'ready', (event: ChartReadyEvent) => this.ready.emit(event));
+    google.visualization.events.addListener(this._controlWrapper, 'error', (event: ChartErrorEvent) => this.error.emit(event));
+    google.visualization.events.addListener(this._controlWrapper, 'statechange', (event: unknown) => this.stateChange.emit(event));
   }
 }

--- a/libs/angular-google-charts/src/lib/components/dashboard/dashboard.component.spec.ts
+++ b/libs/angular-google-charts/src/lib/components/dashboard/dashboard.component.spec.ts
@@ -170,7 +170,7 @@ describe('DashboardComponent', () => {
   describe('ngOnChanges', () => {
     function changeInput<K extends keyof DashboardComponent>(property: K, newValue: DashboardComponent[K]) {
       const oldValue = component[property];
-      component[property as any] = newValue;
+      component[property] = newValue;
       component.ngOnChanges({ [property]: new SimpleChange(oldValue, newValue, oldValue == null) });
     }
 

--- a/libs/angular-google-charts/src/lib/components/dashboard/dashboard.component.ts
+++ b/libs/angular-google-charts/src/lib/components/dashboard/dashboard.component.ts
@@ -41,7 +41,7 @@ export class DashboardComponent implements OnInit, OnChanges {
    * If {@link https://developers.google.com/chart/interactive/docs/roles roles} should be applied, they must be included in this array as well.
    */
   @Input()
-  public columns: Column[];
+  public columns?: Column[];
 
   /**
    * The dashboard has completed drawing and is ready to accept changes.
@@ -61,10 +61,10 @@ export class DashboardComponent implements OnInit, OnChanges {
   public error = new EventEmitter<ChartErrorEvent>();
 
   @ContentChildren(ControlWrapperComponent)
-  private controlWrappers: QueryList<ControlWrapperComponent>;
+  private controlWrappers!: QueryList<ControlWrapperComponent>;
 
-  private dashboard: google.visualization.Dashboard;
-  private dataTable: google.visualization.DataTable;
+  private dashboard?: google.visualization.Dashboard;
+  private dataTable?: google.visualization.DataTable;
   private initialized = false;
 
   constructor(private element: ElementRef, private loaderService: ScriptLoaderService) {}
@@ -77,18 +77,18 @@ export class DashboardComponent implements OnInit, OnChanges {
     });
   }
 
-  public ngOnChanges(changes: SimpleChanges) {
+  public ngOnChanges(changes: SimpleChanges): void {
     if (!this.initialized) {
       return;
     }
 
     if (changes.data || changes.columns) {
       this.createDataTable();
-      this.dashboard.draw(this.dataTable);
+      this.dashboard!.draw(this.dataTable!);
     }
   }
 
-  private createDashboard() {
+  private createDashboard(): void {
     // TODO: This should happen in the control wrapper
     // However, I don't yet know how to do this because then `bind()` would get called multiple times
     // for the same control if something changes. This is not supported by google charts as far as I can tell
@@ -110,22 +110,22 @@ export class DashboardComponent implements OnInit, OnChanges {
     combineLatest([...controlWrappersReady$, ...chartsReady$]).subscribe(() => {
       this.dashboard = new google.visualization.Dashboard(this.element.nativeElement);
       this.initializeBindings();
-      this.dashboard.draw(this.dataTable);
+      this.dashboard.draw(this.dataTable!);
     });
   }
 
-  private initializeBindings() {
+  private initializeBindings(): void {
     this.controlWrappers.forEach(control => {
       if (Array.isArray(control.for)) {
         const chartWrappers = control.for.map(chart => chart.chartWrapper);
-        this.dashboard.bind(control.controlWrapper, chartWrappers);
+        this.dashboard!.bind(control.controlWrapper, chartWrappers);
       } else {
-        this.dashboard.bind(control.controlWrapper, control.for.chartWrapper);
+        this.dashboard!.bind(control.controlWrapper, control.for.chartWrapper);
       }
     });
   }
 
-  private createDataTable() {
+  private createDataTable(): void {
     if (this.data == null) {
       return;
     }

--- a/libs/angular-google-charts/src/lib/components/google-chart/google-chart.component.spec.ts
+++ b/libs/angular-google-charts/src/lib/components/google-chart/google-chart.component.spec.ts
@@ -3,6 +3,7 @@ import { ComponentFixture, fakeAsync, TestBed, tick, waitForAsync } from '@angul
 import { EMPTY, of } from 'rxjs';
 
 import { ChartType } from '../../models/chart-type.model';
+import { ChartReadyEvent } from '../../models/events.model';
 import { ScriptLoaderService } from '../../script-loader/script-loader.service';
 
 import { GoogleChartComponent } from './google-chart.component';
@@ -59,12 +60,13 @@ describe('GoogleChartComponent', () => {
   });
 
   it('should not throw when trying to access chart if its not yet drawn', () => {
+    component['wrapper'] = chartWrapperMock as any;
     expect(() => component.chart).not.toThrow();
   });
 
   describe('chartWrapper', () => {
-    it('should not throw if the chart wrapper is `null`', () => {
-      expect(() => component.chartWrapper).not.toThrow();
+    it('should throw if the chart wrapper is `undefined`', () => {
+      expect(() => component.chartWrapper).toThrow();
     });
 
     it('should return the chart wrapper', () => {
@@ -278,7 +280,7 @@ describe('GoogleChartComponent', () => {
     });
 
     it('should not throw if anything changed but the chart wrapper was not yet initialized', () => {
-      component['wrapper'] = null;
+      component['wrapper'] = undefined;
       component['initialized'] = false;
 
       expect(() => {
@@ -426,7 +428,7 @@ describe('GoogleChartComponent', () => {
       changeInput('dynamicResize', false);
 
       expect(subscriptionMock.unsubscribe).toHaveBeenCalled();
-      expect(component['resizeSubscription']).toBeNull();
+      expect(component['resizeSubscription']).toBeUndefined();
     });
 
     it('should do nothing if the window was resized, but the chart is not yet initialized', fakeAsync(() => {
@@ -499,7 +501,7 @@ describe('GoogleChartComponent', () => {
       component.ngOnInit();
 
       const readySpy = jest.fn();
-      component.ready.subscribe(event => readySpy(event));
+      component.ready.subscribe((event: ChartReadyEvent) => readySpy(event));
 
       const readyCallback = visualizationMock.events.addListener.mock.calls[0][2];
       readyCallback();
@@ -511,7 +513,7 @@ describe('GoogleChartComponent', () => {
       component.ngOnInit();
 
       const errorSpy = jest.fn();
-      component.error.subscribe(event => errorSpy(event));
+      component.error.subscribe((event: ChartReadyEvent) => errorSpy(event));
 
       const errorCallback = visualizationMock.events.addListener.mock.calls[1][2];
 
@@ -528,7 +530,7 @@ describe('GoogleChartComponent', () => {
       chartWrapperMock.getChart.mockReturnValue(chartMock);
 
       const selectSpy = jest.fn();
-      component.select.subscribe(event => selectSpy(event));
+      component.select.subscribe((event: ChartReadyEvent) => selectSpy(event));
 
       component.ngOnInit();
 
@@ -546,7 +548,7 @@ describe('GoogleChartComponent', () => {
 
   function changeInput<K extends keyof GoogleChartComponent>(property: K, newValue: GoogleChartComponent[K]) {
     const oldValue = component[property];
-    component[property as any] = newValue;
+    component[property] = newValue;
     component.ngOnChanges({ [property]: new SimpleChange(oldValue, newValue, oldValue == null) });
   }
 });

--- a/libs/angular-google-charts/src/lib/components/google-chart/google-chart.component.ts
+++ b/libs/angular-google-charts/src/lib/components/google-chart/google-chart.component.ts
@@ -10,7 +10,7 @@ import {
   Output,
   SimpleChanges
 } from '@angular/core';
-import { fromEvent, ReplaySubject, Subscription } from 'rxjs';
+import { fromEvent, Observable, ReplaySubject, Subscription } from 'rxjs';
 import { debounceTime } from 'rxjs/operators';
 
 import { ChartType } from '../../models/chart-type.model';
@@ -60,7 +60,7 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
    * If {@link https://developers.google.com/chart/interactive/docs/roles roles} should be applied, they must be included in this array as well.
    */
   @Input()
-  public columns: Column[];
+  public columns?: Column[];
 
   /**
    * A convenience property used to set the title of the chart.
@@ -127,10 +127,10 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
   @Output()
   public mouseleave = new EventEmitter<ChartMouseLeaveEvent>();
 
-  private dataTable: google.visualization.DataTable;
-  private resizeSubscription: Subscription;
+  private resizeSubscription?: Subscription;
 
-  private wrapper: google.visualization.ChartWrapper;
+  private dataTable: google.visualization.DataTable | undefined;
+  private wrapper: google.visualization.ChartWrapper | undefined;
   private wrapperReadySubject = new ReplaySubject<google.visualization.ChartWrapper>(1);
   private initialized = false;
 
@@ -140,19 +140,19 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
     @Optional() private dashboard?: DashboardComponent
   ) {}
 
-  public get chart(): google.visualization.ChartBase | null {
-    if (!this.wrapper) {
-      return null;
-    }
-
-    return this.wrapper.getChart();
+  public get chart(): google.visualization.ChartBase {
+    return this.chartWrapper.getChart();
   }
 
-  public get wrapperReady$() {
+  public get wrapperReady$(): Observable<google.visualization.ChartWrapper> {
     return this.wrapperReadySubject.asObservable();
   }
 
   public get chartWrapper(): google.visualization.ChartWrapper {
+    if (!this.wrapper) {
+      throw new Error('Trying to access the chart wrapper before it was fully initialized');
+    }
+
     return this.wrapper;
   }
 
@@ -192,17 +192,17 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
       let shouldRedraw = false;
       if (changes.data || changes.columns || changes.formatters) {
         this.createDataTable();
-        this.wrapper.setDataTable(this.dataTable);
+        this.wrapper!.setDataTable(this.dataTable!);
         shouldRedraw = true;
       }
 
       if (changes.type) {
-        this.wrapper.setChartType(this.type);
+        this.wrapper!.setChartType(this.type);
         shouldRedraw = true;
       }
 
       if (changes.options || changes.width || changes.height || changes.title) {
-        this.wrapper.setOptions(this.mergeOptions());
+        this.wrapper!.setOptions(this.mergeOptions());
         shouldRedraw = true;
       }
 
@@ -237,7 +237,7 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
   private updateResizeListener() {
     if (this.resizeSubscription != null) {
       this.resizeSubscription.unsubscribe();
-      this.resizeSubscription = null;
+      this.resizeSubscription = undefined;
     }
 
     if (this.dynamicResize) {
@@ -299,6 +299,6 @@ export class GoogleChartComponent implements ChartBase, OnChanges, OnInit {
       return;
     }
 
-    this.wrapper.draw();
+    this.wrapper!.draw();
   }
 }

--- a/libs/angular-google-charts/src/lib/script-loader/script-loader.service.spec.ts
+++ b/libs/angular-google-charts/src/lib/script-loader/script-loader.service.spec.ts
@@ -18,11 +18,7 @@ describe('ScriptLoaderService', () => {
 
   beforeEach(() => {
     service = TestBed.inject(ScriptLoaderService);
-    globalThis.google = undefined;
-  });
-
-  it('should be created', () => {
-    expect(service).toBeTruthy();
+    (globalThis as any).google = undefined;
   });
 
   describe('isGoogleChartsAvailable', () => {
@@ -166,7 +162,7 @@ describe('ScriptLoaderService', () => {
         expect(chartsMock.load).toHaveBeenCalledWith('current', {
           packages: [],
           language: 'en-US',
-          mapsApiKey: '',
+          mapsApiKey: undefined,
           safeMode: false
         });
       });
@@ -181,7 +177,7 @@ describe('ScriptLoaderService', () => {
         expect(chartsMock.load).toHaveBeenCalledWith('current', {
           packages: [chart],
           language: 'en-US',
-          mapsApiKey: '',
+          mapsApiKey: undefined,
           safeMode: false
         });
       });
@@ -199,7 +195,8 @@ describe('ScriptLoaderService', () => {
 
         expect(loadedSpy).not.toHaveBeenCalled();
 
-        loadCallback();
+        expect(loadCallback!).toBeTruthy();
+        loadCallback!();
         expect(loadedSpy).toHaveBeenCalled();
       });
 

--- a/libs/angular-google-charts/src/lib/script-loader/script-loader.service.ts
+++ b/libs/angular-google-charts/src/lib/script-loader/script-loader.service.ts
@@ -6,7 +6,6 @@ import { GoogleChartsConfig } from '../models/google-charts-config.model';
 import { GOOGLE_CHARTS_CONFIG } from '../models/injection-tokens.model';
 
 const DEFAULT_CONFIG: GoogleChartsConfig = {
-  mapsApiKey: '',
   version: 'current',
   safeMode: false
 };
@@ -14,12 +13,13 @@ const DEFAULT_CONFIG: GoogleChartsConfig = {
 @Injectable({ providedIn: 'root' })
 export class ScriptLoaderService {
   private readonly scriptSource = 'https://www.gstatic.com/charts/loader.js';
-  private readonly scriptLoadSubject = new Subject<void>();
+  private readonly scriptLoadSubject = new Subject<null>();
+  private readonly config: GoogleChartsConfig;
 
   constructor(
     private zone: NgZone,
     @Inject(LOCALE_ID) private localeId: string,
-    @Inject(GOOGLE_CHARTS_CONFIG) @Optional() private config?: GoogleChartsConfig
+    @Inject(GOOGLE_CHARTS_CONFIG) @Optional() config?: GoogleChartsConfig
   ) {
     this.config = { ...DEFAULT_CONFIG, ...(config || {}) };
   }
@@ -49,10 +49,10 @@ export class ScriptLoaderService {
    * @param packages The packages to load.
    * @returns A stream emitting as soon as the chart packages are loaded.
    */
-  public loadChartPackages(...packages: string[]): Observable<void> {
+  public loadChartPackages(...packages: string[]): Observable<null> {
     return this.loadGoogleCharts().pipe(
       switchMap(() => {
-        return new Observable<void>(observer => {
+        return new Observable<null>(observer => {
           const config = {
             packages,
             language: this.localeId,
@@ -60,7 +60,7 @@ export class ScriptLoaderService {
             safeMode: this.config.safeMode
           };
 
-          google.charts.load(this.config.version, config);
+          google.charts.load(this.config.version!, config);
           google.charts.setOnLoadCallback(() => {
             this.zone.run(() => {
               observer.next();
@@ -78,7 +78,7 @@ export class ScriptLoaderService {
    * @returns A stream emitting as soon as loading has completed.
    * If the google charts script is already loaded, the stream emits immediately.
    */
-  private loadGoogleCharts() {
+  private loadGoogleCharts(): Observable<null> {
     if (this.isGoogleChartsAvailable()) {
       return of(null);
     } else if (!this.isLoadingGoogleCharts()) {
@@ -105,7 +105,7 @@ export class ScriptLoaderService {
     return this.getGoogleChartsScript() != null;
   }
 
-  private getGoogleChartsScript(): HTMLScriptElement | null {
+  private getGoogleChartsScript(): HTMLScriptElement | undefined {
     const pageScripts = Array.from(document.getElementsByTagName('script'));
     return pageScripts.find(script => script.src === this.scriptSource);
   }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -10,6 +10,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
+    "strict": true,
     "target": "es2015",
     "typeRoots": ["node_modules/@types"],
     "lib": ["es2017", "dom"],

--- a/tslint.json
+++ b/tslint.json
@@ -12,6 +12,7 @@
     "member-access": true,
     "array-type": [true, "array"],
     "no-host-metadata-property": false,
-    "unified-signatures": false
+    "unified-signatures": false,
+    "no-non-null-assertion": false
   }
 }

